core/src/main/java/brooklyn/entity/basic/Sanitizer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package brooklyn.entity.basic;
+
+import java.util.Map;
+import java.util.Set;
+
+import brooklyn.util.config.ConfigBag;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+public final class Sanitizer {
+
+    private Sanitizer() {
+    } // not instantiable
+
+    public static Map<String, Object> sanitize(ConfigBag input) {
+        return sanitize(input.getAllConfig());
+    }
+
+    public static <K> Map<K, Object> sanitize(Map<K, ?> input) {
+        return sanitize(input, Sets.newHashSet());
+    }
+
+    public static <K> Map<K, Object> sanitize(Map<K, ?> input, Set<Object> visited) {
```

Comment:
Actually I'd build out a non-static implementation whereby the secret detection can be controlled.

    public static Sanitizer(Predicate<Object> sanitizingNeededCheck) { new Sanitizer(sanitizingNeededCheck); }
    protected Sanitizer(Predicate<Object> sanitizingNeededCheck) { ... }
    public <K> Map<K,Object> apply(Map<K,?> input) { return apply(input, Sets.newHashSet()); }
    private <K> Map<K,Object> apply(Map<K,?> input, Set<Object> visitied) {
      // old visited code
    }

then moving `SECRET_NAMES` in becomes a default that `newInstance()` can use, but it can now be customised